core/src/actors/session/handlers.rs

```diff
@@ -55,29 +57,34 @@ impl StreamHandler<BytesMut, Error> for Session {
                         }
                         try_consolidate_session(self, ctx);
                     }
-                    (SessionStatus::Unconsolidated, Command::Verack(_)) => {
+                    // Handler Verack message
+                    (_, SessionStatus::Unconsolidated, Command::Verack(_)) => {
                         handshake_verack(self);
                         try_consolidate_session(self, ctx);
                     }
                     ////////////////////
                     // PEER DISCOVERY //
                     ////////////////////
-                    (SessionStatus::Consolidated, Command::GetPeers(_)) => {
+                    // Handle GetPeers message
+                    (_, SessionStatus::Consolidated, Command::GetPeers(_)) => {
                         peer_discovery_get_peers(self, ctx);
                     }
-                    (SessionStatus::Consolidated, Command::Peers(Peers { peers })) => {
+                    // Handle Peers message
+                    (
+                        SessionType::Outbound,
+                        SessionStatus::Consolidated,
+                        Command::Peers(Peers { peers }),
+                    ) => {
                         peer_discovery_peers(&peers);
                     }
                     /////////////////////
-                    // NOT IMPLEMENTED //
+                    // NOT SUPPORTED   //
                     /////////////////////
-                    (SessionStatus::Consolidated, _) => {
-                        warn!("Not implemented message command received!");
-                    }
-                    (_, kind) => {
+                    (session_type, session_status, msg_type) => {
                         warn!(
-                            "Received a message of kind \"{:?}\", which is not implemented yet",
-                            kind
+                            "Received a message of type \"{:?}\" for a session with {:?} type and\
+                             in {:?} status, which is not implemented yet",
```

Comment:
Just to be in line with the previous "Not supported" comment. :)

```suggestion
                             in {:?} status, which is not supported",
```

I am not 100% sure, but perhaps I would try to shorten the log (to avoid having really long log traces).

```
"Message of type \"{:?}\" for session (type: {:?}, status: {:?}) is not supported"
```
What do you think? :blush:

Reply:
100% agree, I'll change it now, thanks!